packages/utils/src/matchDirectiveValue.ts

```diff
@@ -0,0 +1,12 @@
+export function matchDirectiveValue(dirValue: any, criteria: Record<string, any>) {
+  if (Array.isArray(dirValue)) {
+    return dirValue.some(val => matchesValue(val, criteria));
+  } else if (dirValue) {
+    return matchesValue(dirValue, criteria);
+  }
+  return false;
+}
+
+function matchesValue(dirValue: any, criteria: Record<string, any>) {
+  return Object.keys(criteria).every(key => criteria[key] === dirValue[key]);
```

Comment:
This does not handle nested input object types, there should be a recursive call to matchDirectiveValue somewhere.

This function btw could be called matchValue, not specific to directive.

Suggest test if object vs primitive, if object, recurse, otherwise check primitive if equal